app/src/main/java/com/example/androidtrainingproject/login/LoginScreen.kt

```diff
@@ -47,69 +62,97 @@ fun LoginScreen(navigateToProductDetails: () -> Unit = {}) {
                 .fillMaxWidth()
-                .padding(bottom = 10.dp),
-            value = usernameField.value,
-            label = { Text(text = "Email")},
-            onValueChange = {
-                    newValue -> usernameField.value = newValue
-            }
+                .padding(bottom = dimensionResource(id = R.dimen.padding_sm)),
+            value = usernameField,
+            label = { Text(text = stringResource(id = R.string.label_email)) },
+            colors = TextFieldDefaults.colors(
+                errorIndicatorColor = ErrorDarkRed,
+                errorContainerColor = Color.Transparent,
+                unfocusedContainerColor = Color.Transparent,
+                focusedContainerColor = Color.Transparent,
+            ),
+            isError = markInputsAsWrong,
+            onValueChange = { newValue -> usernameField = newValue }
         )
         OutlinedTextField(
-            modifier = Modifier
-                .padding(bottom = 10.dp)
-                .fillMaxWidth(),
-            value = passwordField.value,
-            label = { Text(text = "Password")},
-            onValueChange = {
-                    newValue -> passwordField.value = newValue
-            },
-            visualTransformation = if (isPasswordVisible.value) VisualTransformation.None else PasswordVisualTransformation(),
+            modifier = Modifier.fillMaxWidth(),
+            value = passwordField,
+            label = { Text(text = stringResource(id = R.string.label_password)) },
+            colors = TextFieldDefaults.colors(
+                errorIndicatorColor = ErrorDarkRed,
+                errorContainerColor = Color.Transparent,
+                unfocusedContainerColor = Color.Transparent,
+                focusedContainerColor = Color.Transparent,
+            ),
+            isError = markInputsAsWrong,
+            onValueChange = { newValue -> passwordField = newValue },
+            visualTransformation = if (isPasswordVisible) VisualTransformation.None else PasswordVisualTransformation(),
             trailingIcon = {
                 IconButton(onClick = {
-                    isPasswordVisible.value = !isPasswordVisible.value
+                    isPasswordVisible = !isPasswordVisible
                 }) {
                     Image(
                         modifier = Modifier
-                            .height(36.dp)
-                            .width(32.dp),
+                            .height(dimensionResource(id = R.dimen.icon_normal))
+                            .width(dimensionResource(id = R.dimen.icon_normal)),
                         painter = painterResource(id = passwordIcon),
-                        contentDescription = "toggle password visibility"
+                        contentDescription = stringResource(id = R.string.toggle_password_visibility)
                     )
                 }
             },
         )
 
         if (loginViewModel.loginFailure) {
```

Comment:
Such logic should always be in a `LaunchedEffect` subscribing to the value in the view model like a `useEffect`, up above the UI logic. Such computational logic shouldn't be in the middle of the UI outside of an effect.